src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ErrorNotificationHandler.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.ExceptionPhase;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.ErrorMsg;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.ErrorNotificationService;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public class ErrorNotificationHandler implements IMessageHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(ErrorNotificationHandler.class);
+
+    private final ErrorNotificationService service;
+
+    private final ObjectMapper mapper;
+
+    private static final Executor SIMPLE_EXEC = Runnable::run;
+
+    private static final Executor SERVICE_EXEC = Executors.newSingleThreadExecutor(r ->
```

Comment:
hope this is all efficient and we may have not threading issues. Although I read this is more efficient 😉